src/core/thread/mesh_forwarder.cpp

```diff
@@ -515,7 +515,7 @@ ThreadError MeshForwarder::UpdateIp6Route(Message &aMessage)
             if (mMle.IsRoutingLocator(ip6Header.GetDestination()))
             {
                 rloc16 = HostSwap16(ip6Header.GetDestination().mFields.m16[7]);
-                VerifyOrExit(mMle.GetRouterId(rloc16) < Mle::kMaxRouterId, error = kThreadError_Drop);
+                VerifyOrExit(mMle.IsRouterIdCorrect(mMle.GetRouterId(rloc16)), error = kThreadError_Drop);
```

Comment:
Since it's a static function, you would only need to do `MleRouter::IsRouterIdCorrect`